### PR TITLE
Use wait_for instead of loop and sleep in docker automation

### PR DESCRIPTION
**Pull request by mirzal**
Followup to #6728 - use `wait_for` instead of loop and sleep in docker automation.

```
$ pytest -v -k 'test_positive_pull_image or end_to_end_pull' tests/foreman/cli/test_docker.py 
===================================================================================== test session starts =====================================================================================
platform linux -- Python 3.6.3, pytest-3.6.1, py-1.7.0, pluggy-0.6.0 -- /home/mzalewsk/.virtualenvs/robottelo/bin/python3                                                                                                                                                

tests/foreman/cli/test_docker.py::DockerClientTestCase::test_positive_container_admin_end_to_end_pull PASSED                                                                            [ 50%]
tests/foreman/cli/test_docker.py::DockerClientTestCase::test_positive_pull_image PASSED                                                                                                 [100%]

========================================================================== 2 passed, 59 deselected in 510.11 seconds ==========================================================================
```

### CODE REVIEW

**Comment by oshtaier** [comment]
@pondrejk from code perspective - it is much cleaner and in general you stop operate with loop iterations, but with time terms directly. Also that method has many nice features like `fail_condition` or `handle_exception` that let you work with results of that method in different ways.

We use it widely in airgun, so you can check there if you like

File: tests/foreman/cli/test_docker.py [hunk 1]
```diff
@@ -1633,15 +1633,16 @@ def test_positive_pull_image(self):
...
-                )
-                if result.return_code == 0:
-                    break
-                sleep(2)
+            result, _ = wait_for(
+                    lambda: ssh.command(
+                            'docker pull {0}'.format(repo['published-at']),
+                            hostname=self.docker_host.ip_addr
+                    ),
+                    num_sec=20 if bz_bug_is_open(1452149) else 1,
```
Comment (oshtaier) [inline tests/foreman/cli/test_docker.py]: 1452149 is in Closed state, so that condition make no sense

File: tests/foreman/cli/test_docker.py [hunk 2]
```diff
@@ -1907,15 +1908,16 @@ def test_positive_container_admin_end_to_end_pull(self):
...
-            )
-            if result.return_code == 0:
-                break
-            sleep(2)
+        result, _ = wait_for(
+                lambda: ssh.command(
+                        docker_pull_command,
+                        hostname=self.docker_host.ip_addr
+                ),
+                num_sec=40 if bz_bug_is_open(1452149) else 1,
```
Comment (oshtaier) [inline tests/foreman/cli/test_docker.py]: same here

File: tests/foreman/cli/test_docker.py (hunk 1 above)
Comment (mirzal) [reply tests/foreman/cli/test_docker.py]: Yes, but it's Closed Wontfix, meaning the problem that prompted us to add this condition has not been properly resolved. Thought this function is able to tell the difference.

I'll just set it unconditionally, then. Unless you know better solution?

**Pull request merged**